ICSharpCode.Decompiler/Metadata/DotNetCorePathFinder.cs

```diff
@@ -249,21 +249,22 @@ public string TryResolveDotNetCoreShared(IAssemblyReference name, out string run
 		static string GetClosestVersionFolder(string basePath, Version version)
 		{
 			var foundVersions = new DirectoryInfo(basePath).GetDirectories()
-				.Select(d => ConvertToVersion(d.Name))
+				.Select(ConvertToVersion)
 				.Where(v => v.version != null);
 			foreach (var folder in foundVersions.OrderBy(v => v.version))
 			{
 				if (folder.version >= version)
-					return folder.directoryName;
+					if(folder.directory.EnumerateFiles().Any())
```

Comment:
It seems that your changes broke the reference assembly resolver logic that is used by our tests. The structure of reference assembly folder is slightly different. A possible fix would be:

				if (folder.version >= version
					&& folder.directory.EnumerateFiles("*.dll", SearchOption.AllDirectories).Any())
				{
					return folder.directory.Name;
				}

This way we search for libraries in sub-folders as well.

Reply:
I adjusted this part to your suggestion.